dhcpv6/option_vendor_opts.go

```diff
@@ -0,0 +1,87 @@
+package dhcpv6
+
+/*
+ This module defines the OptVendorOpts structure.
+ https://tools.ietf.org/html/rfc3315#section-22.17
+
+       0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+      |      OPTION_VENDOR_OPTS       |           option-len          |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+      |                       enterprise-number                       |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+      .                                                               .
+      .                          option-data                          .
+      .                                                               .
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+*/
+
+import (
+	"encoding/binary"
+	"fmt"
+)
+
+// OptVendorOpts represents a DHCPv6 Status Code option
+type OptVendorOpts struct {
+	enterpriseNumber uint32
+	vendorOpts       []byte
```

Comment:
I will do some testing and see if this works with what the vendor is sending us. I don't think they are sending the data as additional options in that field.

Reply:
feel free to ping me privately for sake of speed

Reply:
> I've been testing with the GenericOption since the last comment here but now it seems to be grabbing too many bytes by default for what the vendor is now sending me.

Are all of those bytes part of the vendor options?

> Would you recommend I request the vendor to change the option 17 data they are sending me to better match the RFC

Yes, ideally they should be following the RFC.

Reply:
> according to the RFC this is a list of DHCP option fields, so this can become []Option. This will also help enforce the correct format and size

I had a look at this with @gorhamc. As per RFC I think we should treat them as opaque data instead of `[]Option`. Same thing that we are doing for option 16.

Reply:
I spoke to soon. They have the same format as an `Option` but the codes do not match the ones in iana so maybe we can treat all of them as generic options.